docs/reference/tab-widgets/troubleshooting/data/increase-cluster-shard-limit.asciidoc

```diff
@@ -0,0 +1,210 @@
+//////////////////////////
+
+[source,console]
+--------------------------------------------------
+PUT my-index-000001
+
+--------------------------------------------------
+// TESTSETUP
+
+[source,console]
+--------------------------------------------------
+PUT _cluster/settings
+{
+  "persistent" : {
+    "cluster.routing.allocation.total_shards_per_node" : null
+  }
+}
+
+DELETE my-index-000001
+--------------------------------------------------
+// TEARDOWN
+
+//////////////////////////
+
+// tag::cloud[]
+In order to get the shards assigned we'll need to increase the number of shards 
+that can be collocated on a node in the cluster in order to get all shards assigned. 
+We'll achieve this by inspecting the system-wide `cluster.routing.allocation.total_shards_per_node` 
+<<cluster-get-settings, cluster setting>> and increasing the configured value.
+
+Note that this requires the current data nodes to have enough disk space to host 
+all the data.
+If the current deployment configuration doesn't have enough capacity to store
+all the data, including the unassigned shards, please contact 
+https://support.elastic.co[Elastic Support] to discuss available options.
```

Comment:
Couldn't they add nodes to the cluster to help this? Should we go directly to contacting support?

Reply:
Not in cloud. Gotta be GREEN before scaling up unfortunately.

Reply:
Couldn't they alternatively delete indices? Wouldn't that be the safer option if it's possible?

Reply:
That's an option as well. I'm not sure if we should send users that way though if there are alternatives via support. @Leaf-Lin what do you think?

Reply:
The setting `cluster.routing.allocation.total_shards_per_node` by default is (-1) unlimited.  The setting may only be adjusted in order to balance shard counts between nodes, although the [recommended way](https://www.elastic.co/guide/en/elasticsearch/reference/current/size-your-shards.html#avoid-node-hotspots) is to adjust _index_ setting `index.routing.allocation.total_shards_per_node`, not the _cluster_ setting.  

From my point of view, if for some reason, users adjusted this setting which causes some shards to be unassigned, we should simply ask user to inspect the setting and increase the configured value as stated in the troubleshooting doc.  

I would probably remove "contact support" in this particular scenario.

----

Note, this is different from the [`cluster.max_shards_per_node` setting](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-cluster.html#cluster-max-shards-per-node) which is 1000 by default and should be resolved by adding nodes or delete data as the comments suggested.  When the `max_shards_per_node` limit is reached, this may cause additional resource (heap) to be consumed on that particular node, thus additional resources are required (or data should be removed) to resolve the situation.

Reply:
To close this loop - total_shards_per_node are usually configured in order to control the shard layout distribution and increasing the value should always be possible.

Disk usage issues will be reported by a different indicator (disk usage indicator probably) and deleting data would be advised there if everything else is not possible.